lib/_emerge/depgraph.py

```diff
@@ -9172,9 +9174,17 @@ def gather_deps(ignore_priority, mergeable_nodes, selected_nodes, node):
                 # (these occur when new portage blocks an older package version).
                 return False
             selected_nodes.add(node)
+            if smallest_cycle is not None and len(selected_nodes) >= len(
```

Comment:
I've not read over this yet outside of the gh viewer so it's harder to see the surrounding stuff - but I wonder if we should do something like store a tuple of `(ignore_priority, size)` or maybe even `(greatest_priority_ignored, size)` and go after that instead. (It might be that this isn't relevant because we don't take the path in this case, so ignore me if so)

Reply:
I'm not sure how the ignore_priority part of this tuple is useful. The ignore_priority objects are not really comparable to each other, so I'm not sure what you would want to do with it.

Reply:
The idea would be that if we had `ignore_priority` set for a cycle solution, we'd prefer one unset unless we've tried for a while and can't get anything better?

Reply:
Maybe I'm trying to reinvent more fine-grained priority stuff actually and we shouldn't worry here.

Reply:
It starts with local_priority_range.MEDIUM_POST and only increases ignore_priority from there. When find_smallest_cycle is called, we assume that we need to ignore something, since otherwise it would have selected something earlier here:
```python
                for i in range(priority_range.NONE, priority_range.MEDIUM_SOFT + 1):
                    ignore_priority = priority_range.ignore_priority[i]
                    nodes = get_nodes(ignore_priority=ignore_priority)
```

Reply:
Got it, thank you!